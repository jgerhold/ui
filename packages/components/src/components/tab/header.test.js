import { mount } from '@vue/test-utils'

import TabBody from './Body'

describe('TabBody', () => {
  describe('component', () => {
    test('should be a Vue instance', async () => {
      const wrapper = await mount(TabBody, {
        slots: {
          default: '<div>Inner Tab</div>'
        }
      })

      expect(wrapper.isVueInstance()).toBeTruthy()
    })
  })

  describe('props', () => {
    test('should render without props', async () => {
      const wrapper = await mount(TabBody, {
        slots: {
          default: '<div>Inner Tab</div>'
        }
      })

      expect(wrapper.element).toMatchSnapshot()
    })

    const props = [
      {
        prop: 'active',
        value: true
      },
      {
        prop: 'name',
        value: 'tab-name'
      },
      {
        prop: 'background',
        value: '#fff'
      },
      {
        prop: 'fixed',
        value: true
      }
    ]

    props.forEach(({ prop, value }) => {
      test(`should render with prop ${prop}`, async () => {
        const wrapper = await mount(TabBody, {
          slots: {
            default: '<div>Inner Tab</div>'
          },
          propsData: {
            [prop]: value
          }
        })

        expect(wrapper.element).toMatchSnapshot()
      })
    })
  })
})
