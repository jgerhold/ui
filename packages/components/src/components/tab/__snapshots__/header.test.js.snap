// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBody props should render with prop active 1`] = `
<div
  aria-labelledby="trigger-"
  class="tab-body active"
  role="tabpanel"
  tabindex="0"
>
  <div
    class="tab-content"
    style="background-color: rgb(43, 138, 198);"
  >
    <div>
      Inner Tab
    </div>
  </div>
</div>
`;

exports[`TabBody props should render with prop background 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="trigger-"
  class="tab-body"
  role="tabpanel"
  tabindex="0"
>
  <div
    class="tab-content"
    style="background-color: rgb(255, 255, 255);"
  >
    <div>
      Inner Tab
    </div>
  </div>
</div>
`;

exports[`TabBody props should render with prop fixed 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="trigger-"
  class="tab-body fixed"
  role="tabpanel"
  tabindex="0"
>
  <div
    class="tab-content"
    style="background-color: rgb(43, 138, 198);"
  >
    <div>
      Inner Tab
    </div>
  </div>
</div>
`;

exports[`TabBody props should render with prop name 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="trigger-tab-name"
  class="tab-body"
  role="tabpanel"
  tabindex="0"
>
  <div
    class="tab-content"
    style="background-color: rgb(43, 138, 198);"
  >
    <div>
      Inner Tab
    </div>
  </div>
</div>
`;

exports[`TabBody props should render without props 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="trigger-"
  class="tab-body"
  role="tabpanel"
  tabindex="0"
>
  <div
    class="tab-content"
    style="background-color: rgb(43, 138, 198);"
  >
    <div>
      Inner Tab
    </div>
  </div>
</div>
`;
